src/python/pants/backend/python/lint/ruff/subsystem.py

```diff
@@ -83,6 +105,45 @@ def config_request(self, dirs: Iterable[str]) -> ConfigFilesRequest:
             check_content={os.path.join(d, "pyproject.toml"): b"[tool.ruff" for d in all_dirs},
         )
 
+    # Options that only exist to ease the upgrade from Ruff as a Python tool to Ruff as an external
+    # downloaded one
+    install_from_resolve = StrOption(
+        advanced=True,
+        default=None,
+        removal_version="2.26.0.dev0",
+        removal_hint="Now ignored: use `version` and `known_version` to customise the version, consider deleting the resolve and `python_requirement` if no longer used",
```

Comment:
Just so we're talking about the same thing: are you questioning whether we include these at all, or questioning using `2.26.0.dev0` as the removal version?

Reply:
> questioning using 2.26.0.dev0 as the removal version?

This one. 

I understand why it's included as it provides a smoother upgrade path for the user. But given that the old resolve-based functionality will no longer work, I don't see the point in leaving this in for 2 releases as the options are ignored. I think it makes sense to give 2 releases for deprecation when the old functionality still works to allow users to migrate. But since this is a breaking change, IMO we should force users to migrate to the new options ASAP.

Reply:
Ah okay. I see where you're coming from.

I generally prefer longer deprecations: if we remove the options, it's harder for someone to do pre-release testing, and making prerelease testing as easy as possible is valuable IMO.

(Spelling out an example: if we removed the flags in 2.24, someone might be on 2.22, and then cannot do pre-release testing of any features of 2.24.0.devX releases (even non-ruff ones) until they fix the `[ruff]` part of `pants.toml`.)

That said, having the fields around in 2.25 in addition to 2.23 and 2.24 might be somewhat unnecessary, although it gives us a bit more flexibility/slack in terms of which release branches are live/stable when.

From my perspective _forcing_ by removing the options seems to have less value:

- if someone upgrades Pants and has Ruff's behaviour change due to the change in version, hopefully they'd notice that because CI fails (or whatever), and that'd prompt them to pay attention to the warning, hopefully. 
- if someone upgrades Pants and the behaviour does NOT change, that seems acceptable, and the user can notice and deal with the warning on their own time.

(Plus, the diagnostics being bad for an option that's fully removed, i.e. it's harder for someone to update in that case.)

I'll iterate on the message a bit:

```
$ MODE=debug PANTS_SOURCE=~/projects/pantsbuild/pants pants lint --only=ruff-check ::

...

12:57:53.58 [WARN] DEPRECATED: option 'install_from_resolve' in scope 'ruff' is scheduled to be removed in version 2.26.0.dev0.

NOW IGNORED: use `version` and `known_version` options to customise the version of ruff, replacing this option; consider deleting the resolve and `python_requirement` if no longer used. See https://www.pantsbuild.org/2.23/reference/subsystems/ruff
12:57:54.51 [INFO] Completed: Lint with `ruff check` - ruff check succeeded.
All checks passed!

...
```

Thoughts?

Reply:
I buy it - I think pre-release testing is a valuable use case so we can leave it in.